vllm/v1/core/scheduler.py

```diff
@@ -125,6 +174,12 @@ def schedule(self) -> "SchedulerOutput":
             assert not has_partial_request
             assert token_budget > 0
             request = self.running[req_index]
+
+            # Skip requests waiting for FSM
+            if request.status == RequestStatus.WAITING_FOR_FSM:
```

Comment:
Why would this be true? Aren't requests with guided decoding first put into the `grammar` queue, and then only added to the waiting -> running queues once the FSM has been created?